org.lflang.tests/src/org/lflang/tests/TestBase.java

```diff
@@ -426,48 +456,163 @@ private void generateCode(LFTest test) {
+            }
+        }
+        return fedNameToDockerFile;
+    }
+
+    /**
+     * Return the content of the bash script used for testing docker option in federated execution. 
+     * @param fedNameToDockerFile A mapping of federateName -> absolutePathToDockerFile
+     * @param testNetworkName The name of the network used for testing the docker option. 
+     *                        See https://github.com/lf-lang/lingua-franca/wiki/Containerized-Execution#federated-execution for more details.
+     */
+    private String getDockerRunScript(Map<String, Path> fedNameToDockerFile, String testNetworkName) {
+        StringBuilder shCode = new StringBuilder();
+        shCode.append("#!/bin/bash\n");
+        int n = fedNameToDockerFile.size();
+        shCode.append("pids=\"\"\n");
+        shCode.append(String.format("docker run --rm --network=%s --name=rti rti:test -i 1 -n %d &\n", testNetworkName, n));
+        shCode.append("pids+=\"$!\"\nsleep 3\n");
+        for (String fedName : fedNameToDockerFile.keySet()) {
+            Path dockerFile = fedNameToDockerFile.get(fedName);
+            shCode.append(String.format("docker run --rm --network=%s %s:test -i 1 &\n", testNetworkName, fedName));
+            shCode.append("pids+=\" $!\"\n");
+        }
+        shCode.append("for p in $pids; do\n");
+        shCode.append("    if wait $p; then\n");
+        shCode.append("        :\n");
+        shCode.append("    else\n");
+        shCode.append("        exit 1\n");
+        shCode.append("    fi\n");
+        shCode.append("done\n");
+        return shCode.toString();
+    }
+
+    /**
+     * Return a list of ProcessBuilders used to test the docker option under non-federated execution.
+     * @param test The test to get the execution command for.
+     */
+    private List<ProcessBuilder> getNonfederatedDockerExecCommand(LFTest test) {
+        var srcGenPath = test.fileConfig.getSrcGenPath();
+        var dockerPath = srcGenPath.resolve(test.fileConfig.name + ".Dockerfile");
+        return Arrays.asList(new ProcessBuilder("docker", "build", "-t", "lingua_franca:test", "-f", dockerPath.toString(), srcGenPath.toString()), 
+                             new ProcessBuilder("docker", "run", "--rm", "lingua_franca:test"),
+                             new ProcessBuilder("docker", "image", "rm", "lingua_franca:test"));
+    }
+
+    /**
+     * Return a list of ProcessBuilders used to test the docker option under federated execution.
+     * @param test The test to get the execution command for.
+     */
+    private List<ProcessBuilder> getFederatedDockerExecCommand(LFTest test) {
+        var rtiPath = test.fileConfig.getSrcGenBasePath().resolve("RTI");
+        var rtiDockerPath = rtiPath.resolve("rti.Dockerfile");
+        Map<String, Path> fedNameToDockerFile = getFederatedDockerFiles(test);
+        try {
+            File testScript = File.createTempFile("dockertest", null);
```

Comment:
I think that the contents of this function can be made more compact by creating a list that contains the RTI as well as all the federates. Also, perhaps instead of creating a bash script, the execution commands could be appended to the list of commands? If the need for `sleep 3` is a concern, we should probably address the underlying problem, where the RTI must be up and running before federates can execute.

Reply:
Actually, I wonder if this type of testing would become redundant when we have `docker compose` tests (and figure out how to read the return value)? Maybe we should leave a FIXME here to come back to it later? That would make the workflow of this testing a lot cleaner perhaps.

Reply:
`sleep 3` is an arbitrary number. I agree that testing with docker compose would be simpler. I'll look into that.

Reply:
I think it is best to add docker-compose testing in a PR branched from #754. I'll mark this as resolved for now